helper/schema/resource_diff.go

```diff
@@ -378,6 +378,25 @@ func (d *ResourceDiff) GetOk(key string) (interface{}, bool) {
 	return r.Value, exists
 }
 
+// GetOkExists functions the same way as GetOkExists within ResourceData, but
+// it also checks the new diff levels to provide data consistent with the
+// current state of the customized diff.
+func (d *ResourceDiff) GetOkExists(key string) (interface{}, bool) {
+	r := d.get(strings.Split(key, "."), "newDiff")
+	exists := r.Exists && !r.Computed
+	return r.Value, exists
+}
+
+// Computed exposes the computed value of a field read result to the caller.
```

Comment:
This phrasing of the documentation confused me a little at first until I read and understood the implementation. I'd suggest rewording it to describe the effect without discussing the implementation detail of a "field read result", such as:

> Computed returns true if the new value for the given key is marked as computed in the diff.

In the new HCL2-based model we've been gradually making a subtle distinction between "computed" (which describes the general behavior of a field regardless of its current state) and "unknown"/"known" (which describe the current state of the computed field's value). So far that distinction hasn't really affected the public-facing `helper/schema` APIs much since we've not generally exposed this idea to providers, but maybe this is an opportunity to get ahead of this by calling this method `NewValueKnown`, or something similar.